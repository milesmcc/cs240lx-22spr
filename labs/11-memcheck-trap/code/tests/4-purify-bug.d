tests/4-purify-bug.o: tests/4-purify-bug.c \
 /Users/miles/Source/cs240lx/libpi/include/rpi.h \
 /Users/miles/Source/cs240lx/libpi/include/gpio.h \
 /Users/miles/Source/cs240lx/libpi/include/../libc/libc-extra.h \
 /Users/miles/Source/cs240lx/libpi/include/../libc/assert.h \
 /Users/miles/Source/cs240lx/libpi/include/rpi-constants.h \
 /Users/miles/Source/cs240lx/libpi/include/rpi-extra.h purify.h \
 /Users/miles/Source/cs240lx/libpi/include/src-loc.h memtrace.h \
 /Users/miles/Source/cs240lx/libpi/include/procmap.h full-except.h \
 /Users/miles/Source/cs240lx/libpi/include/vector-base.h \
 /Users/miles/Source/cs240lx/libpi/libc/bit-support.h \
 /Users/miles/Source/cs240lx/libpi/include/asm-helpers.h
