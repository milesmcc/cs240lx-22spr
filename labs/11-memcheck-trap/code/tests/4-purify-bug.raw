opened tty port </dev/cu.usbserial-0001>.
my-install: tty-usb=</dev/cu.usbserial-0001> program=<./tests/4-purify-bug.bin> about to boot
TRACE:simple_boot: sending 30084 bytes, crc32=abe8644f
waiting for a start...
PRINT_STRING:pi sent print: <glen3b: success: Received the program!>
bootloader: Done.
TRACE:should detect use after free
TRACE:ERROR:Allocated block 1 [0x183bc8] corrupted at offset -4
TRACE:	logical block id=1, [addr=0x183bc8] nbytes=824
TRACE:	Block allocated at: tests/4-purify-bug.c:notmain:17
TRACE:ERROR:Allocated block 1 [0x183bc8] corrupted at offset -3
TRACE:	logical block id=1, [addr=0x183bc8] nbytes=824
TRACE:	Block allocated at: tests/4-purify-bug.c:notmain:17
PANIC:../../labs/9-memcheck-stat/code.staff/ckalloc.c:ckalloc:150:mem_check(h, ck_get_rz1(h), rn)
DONE!!!

bootloader: pi exited.  cleaning up
