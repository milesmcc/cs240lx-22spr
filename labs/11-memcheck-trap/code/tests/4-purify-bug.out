TRACE:should detect use after free
TRACE:ERROR:Allocated block 1  corrupted at offset -4
TRACE:	logical block id=1,  nbytes=824
TRACE:	Block allocated at: tests/4-purify-bug.c:notmain:17
TRACE:ERROR:Allocated block 1  corrupted at offset -3
TRACE:	logical block id=1,  nbytes=824
TRACE:	Block allocated at: tests/4-purify-bug.c:notmain:17
PANIC:../../labs/9-memcheck-stat/code.staff/ckalloc.c:ckalloc:150:mem_check(h, ck_get_rz1(h), rn)
